### PR TITLE
Add starfish v1 attributes to span data/breadcrumbs

### LINKED ISSUE #765: [Starfish] Add db attributes to database span's span data
To get an overview of this work across all SDKs, please visit:
https://github.com/orgs/getsentry/projects/135

Parent issue: https://github.com/getsentry/team-sdks/issues/19

For the starfish db module it was determined that we need extra database attributes, specifically about the database connection attributes.

We're going to matching the [otel db conventions](https://opentelemetry.io/docs/specs/otel/trace/semantic_conventions/database/). Either one of `server.address` or `server.socket.address` is required. If `server.address` is set, then `server.port` should be set. If `server.port` is different than `server.socket.port`, than `server.socket.port` should also be set. In addition, `db.name` should be set if possible. `db.system` should match the list in [OpenTelemetry's well known conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/database.md#notes-and-well-known-identifiers-for-dbsystem).

| Attribute  | Type | Description  | Examples  | Requirement Level |
|---|---|---|---|---|
| `db.system` | string | An identifier for the database management system (DBMS) product being used. | `mysql` | Required |
| `server.address` | string | Name of the database host. | `example.com` | Conditionally Required: See alternative attributes below. |
| `server.port` | int | Logical server port number | `80`; `8080`; `443` | Conditionally Required: [1] |
| `server.socket.address` | string | Physical server IP address or Unix socket address. | `10.5.3.2` | See below |
| `server.socket.port` | int | Physical server port. | `16456` | Recommended: If different than `server.port`. |
| `db.name` | string | This attribute is used to report the name of the database being accessed. For commands that switch the database, this should be set to the target database (even if the command fails). | `customers`; `main` | Conditionally Required: If applicable. |

These values should be set on `span.data`.

### Motivation

This is being done for the Sentry's [new performance views for databases](https://sentry.sentry.io/starfish/database/). See a WIP example of the view below:

image

### PR DESCRIPTION
**Pull request by cleptric**
Closes #765 

See https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/trace/semantic_conventions

### CODE REVIEW

File: src/Sentry/Laravel/Tracing/EventHandler.php [hunk 1]
```diff
@@ -182,14 +182,22 @@ protected function queryExecutedHandler(DatabaseEvents\QueryExecuted $query): vo
         $context = new SpanContext();
         $context->setOp('db.sql.query');
         $context->setDescription($query->sql);
+        $context->setData([
+            'db.name' => $query->connection->getDatabaseName(),
+            'db.system' => $query->connection->getDriverName(),
+            // 'server.address' => ,
+            // 'server.port'=> ,
```
Comment (cleptric) [inline src/Sentry/Laravel/Tracing/EventHandler.php:189]: @stayallive any idea how we can elegantly fetch these values?

We would also need to distinguish between TCP and socket connections, `server.address` vs `server.socket.address`.

File: src/Sentry/Laravel/EventHandler.php [hunk 2]
```diff
@@ -287,12 +287,12 @@ protected function httpClientResponseReceivedHandler(HttpClientEvents\ResponseRe
             null,
             [
                 'url' => $this->getPartialUri($fullUri),
```
[the remark this answers is not in the document]
Comment (AbhiPrasad) [reply src/Sentry/Laravel/EventHandler.php:289]: for now yes

File: src/Sentry/Laravel/Tracing/EventHandler.php (hunk 1 above)
Comment (stayallive) [reply src/Sentry/Laravel/Tracing/EventHandler.php:189]: Does it need to be elegant though... I think we can probably coorce the driver to give us those details. Is there also a way to say it's a file? For sqlite for example?

Comment (cleptric) [reply src/Sentry/Laravel/Tracing/EventHandler.php:189]: If it works, it works 🪨😬

@AbhiPrasad about the sqlite case

Comment (AbhiPrasad) [reply src/Sentry/Laravel/Tracing/EventHandler.php:189]: We can forgo `server.address` if it's just a file IMO

**Comment by mfb** [comment]
Are there also standards for how http.server transactions are tagged? I noticed that laravel and symfony do it a bit differently.

**Comment by AbhiPrasad** [comment]
> Are there also standards for how http.server transactions are tagged

We have some, but they are sparse. If you create a GH issue with the inconsistencies we can take a look and audit.

**Comment by mfb** [comment]
> We have some, but they are sparse. If you create a GH issue with the inconsistencies we can take a look and audit.

Not sure where to open an issue, but in https://github.com/getsentry/sentry-laravel/blob/f8685aac8880701628bf24848b0c35f09b6888f3/src/Sentry/Laravel/Tracing/Middleware.php#L177 `url` and `method` are set as data, while in https://github.com/getsentry/sentry-symfony/blob/3e9e23da8985818872cf91e5d5d59895a1421154/src/EventListener/TracingRequestListener.php#L89 `http.url` and `http.method` are set as tags.

**Comment by cleptric** [comment]
Moving forward, we'll remove tags set by the SDK and rely on request/response context and span data instead.

**Review by AbhiPrasad — APPROVED** [review]
:shipit:

**Pull request merged**